Userland/Libraries/LibJS/Parser.h

```diff
@@ -107,15 +107,26 @@ class Parser {
         {
             if (!position.has_value())
                 return {};
-            // We need to modify the source to match what the lexer considers one line - normalizing
-            // line terminators to \n is easier than splitting using all different LT characters.
-            String source_string { source };
-            source_string.replace("\r\n", "\n");
-            source_string.replace("\r", "\n");
-            source_string.replace(LINE_SEPARATOR, "\n");
-            source_string.replace(PARAGRAPH_SEPARATOR, "\n");
+
             StringBuilder builder;
-            builder.append(source_string.split_view('\n', true)[position.value().line - 1]);
+            String source_string { source };
+            GenericLexer lexer(source_string);
```

Comment:
I know this was already merged, but is there a reason this can't avoid a string creation?
```c++
GenericLexer lexer(source);
```